tests/config/run_test.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+set -xe
+
+# Check for a command-line argument
+if [ $# -eq 0 ]; then
+  echo "Error: Please provide a base path as a command-line argument."
+  exit 1
+fi
+
+function clean() {
+    rm -f "${base_path}/conf_test.conf"
+    rm -f "${base_path}/clvk.conf"
+}
+trap clean EXIT
+
+# Assign the base path from the argument
+base_path=$1
+
+# Create the first file with the base path
+echo "cache_dir=testing" > "${base_path}/conf_test.conf"
+
+# Create the second file with the base path
+echo "cache_dir=failed" > "${base_path}/clvk.conf"
+echo "compiler_temp_dir=not/overwritten/" > "${base_path}/clvk.conf"
```

Comment:
are you sure that this will be the current directory?
Maybe do something like:
```
TEMP_DIR="$(mktemp -d)"
pushd "${TEMP_DIR}"

...
[ create the files & execute the test ]
...

popd
 ```

Reply:
Note that it would also mean, you would not need base_path I believe

Reply:
I think I will still need the binary path for the config_test binary